scripts/uploaded-files/record-existing-files.ts

```diff
@@ -122,30 +62,56 @@ const recordFile = async (
   }
 };
 
+/**
+ * Formats the field name to be used in SQL queries, handling nested JSON fields.
+ * Examples:
+ * - "url" will be converted to "url"
+ * - "Collective.settings.customEmailMessage" will be converted to "settings"->>'customEmailMessage'
+ * - "Collective.settings.nested.message" will be converted to "settings"->'nested'->>'message'
+ */
+const prepareFieldNameForSQL = (fieldName: string): string => {
+  const splitName = fieldName.split('.');
+  return splitName
+    .map((name, index) => {
+      if (index === 0) {
+        return `"${name}"`;
+      } else if (index === splitName.length - 1) {
+        return `->>'${name}'`;
+      } else {
+        return `->'${name}'`;
+      }
+    })
+    .join('');
+};
+
```

Comment:
I guess we dont need to worry too much about escaping here, since its input is hard coded from the fields constants right?

The `column#>{path}` operator is useful for json path access, simplifying it from [sequelize](https://github.com/sequelize/sequelize/blob/f6f8ba759271c8eff4acdcabcbb44a2d82b00419/packages/core/src/dialects/postgres/query-generator.js#L982):

```
const prepareFieldNameForSQL = (fieldName: string): string => {
  const [column, ...path] = _.toPath(fieldName);

  return `${column}#>>{${path.join(',')}}`;
};
```

Reply:
On second thought, for future reference maybe it deserves a comment about not escaping input to avoid copying it to another context.

Reply:
Good call